worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java

```diff
@@ -0,0 +1,274 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.worker.ingestion;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.cafdataprocessing.services.staging.BatchId;
+import com.github.cafdataprocessing.services.staging.TenantId;
+import com.github.cafdataprocessing.services.staging.dao.filesystem.BatchPathProvider;
+import com.github.cafdataprocessing.services.staging.exceptions.InvalidBatchIdException;
+import com.github.cafdataprocessing.services.staging.exceptions.InvalidTenantIdException;
+import com.github.cafdataprocessing.worker.ingestion.models.Subbatch;
+import com.hpe.caf.worker.batch.BatchDefinitionException;
+import com.hpe.caf.worker.batch.BatchWorkerPlugin;
+import com.hpe.caf.worker.batch.BatchWorkerServices;
+import com.hpe.caf.worker.document.DocumentWorkerConstants;
+import com.hpe.caf.worker.document.DocumentWorkerDocumentTask;
+import com.hpe.caf.worker.document.DocumentWorkerScript;
+import java.io.File;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Plugin that expects a batch definition as a String of IDs and will create tasks from these IDs.
+ *
+ * The format could be,
+ *
+ * tenantId/batch1|batch2|batch3|... 
+ * tenantId/batch1 
+ * subbatch:tenantId/batchId/file-json.batch
+ *
+ */
+@Slf4j
+public class IngestionBatchWorkerPlugin implements BatchWorkerPlugin
+{
+    private final ObjectMapper mapper;
+    private final BatchPathProvider fileSystemProvider;
+
+    public IngestionBatchWorkerPlugin()
+    {
+        final String env = System.getenv("CAF_STAGING_SERVICE_BASEPATH");
+        if (StringUtils.isEmpty(env)) {
+            throw new RuntimeException("CAF_STAGING_SERVICE_BASEPATH environment variable not set");
+        }
+        fileSystemProvider = new BatchPathProvider(env);
+        mapper = new ObjectMapper();
+    }
+
+    @Override
+    public void processBatch(final BatchWorkerServices batchWorkerServices, final String batchDefinition,
+                             final String taskMessageType, final Map<String, String> taskMessageParams)
+        throws BatchDefinitionException
+    {
+        log.debug("Definitions received: " + batchDefinition);
+        if (taskMessageParams != null) {
+            log.debug("Task message params: " + taskMessageParams.toString());
+        }
+
+        if (batchDefinition == null || batchDefinition.trim().length() == 0) {
+            log.error("IngestionBatchWorkerPlugin has not received a valid batch definition string");
+            throw new RuntimeException("IngestionBatchWorkerPlugin has not received a valid batch definition string");
```

Comment:
This should be a `BatchDefinitionException`.  That is the whole point of it - there is a problem with the batch definition.